### PR TITLE
[1.10.X] Add Block#getSound

### PR DESCRIPTION
**Pull request by Madsthunder**
All this does is creates a new method Block#getSound that gets a SoundEvent based on the EnumSoundType provided

### CODE REVIEW

File: patches/minecraft/net/minecraft/block/Block.java.patch [hunk 1]
```diff
@@ -1,6 +1,14 @@
 --- ../src-base/minecraft/net/minecraft/block/Block.java
 +++ ../src-work/minecraft/net/minecraft/block/Block.java
-@@ -44,11 +44,12 @@
+@@ -32,6 +32,7 @@
+ import net.minecraft.util.ObjectIntIdentityMap;
+ import net.minecraft.util.ResourceLocation;
+ import net.minecraft.util.Rotation;
++import net.minecraft.util.SoundEvent;
```
Comment (shadowfacts) [inline patches/minecraft/net/minecraft/block/Block.java.patch]: No imports in patches

File: patches/minecraft/net/minecraft/block/Block.java.patch [hunk 2]
```diff
@@ -195,7 +203,24 @@
      }
  
      @SideOnly(Side.CLIENT)
-@@ -893,6 +912,1141 @@
+@@ -888,11 +908,1158 @@
+         return this.field_149762_H;
+     }
+ 
++    public SoundEvent getSound(IBlockAccess worldIn, BlockPos pos, Entity entity, net.minecraftforge.common.EnumSoundType soundType)
```
Comment (shadowfacts) [inline patches/minecraft/net/minecraft/block/Block.java.patch]: JavaDoc?

File: patches/minecraft/net/minecraft/client/multiplayer/PlayerControllerMP.java.patch [hunk 3]
```diff
@@ -1,6 +1,32 @@
 --- ../src-base/minecraft/net/minecraft/client/multiplayer/PlayerControllerMP.java
 +++ ../src-work/minecraft/net/minecraft/client/multiplayer/PlayerControllerMP.java
-@@ -123,6 +123,12 @@
+@@ -1,7 +1,8 @@
+ package net.minecraft.client.multiplayer;
+ 
+-import io.netty.buffer.Unpooled;
```
Comment (shadowfacts) [inline patches/minecraft/net/minecraft/client/multiplayer/PlayerControllerMP.java.patch]: Again, no changing imports in patches

File: patches/minecraft/net/minecraft/item/ItemBed.java.patch [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+ import net.minecraft.util.EnumHand;
+ import net.minecraft.util.SoundCategory;
++import net.minecraft.util.SoundEvent;
+ import net.minecraft.util.math.BlockPos;
+ import net.minecraft.util.math.MathHelper;
+ import net.minecraft.world.World;
+ 
+ public class ItemBed extends Item
+ {
+-    public ItemBed()
```
Comment (shadowfacts) [inline patches/minecraft/net/minecraft/item/ItemBed.java.patch]: You changed the indentation for the constructor...

File: patches/minecraft/net/minecraft/item/ItemBed.java.patch [hunk 5]
```diff
@@ -0,0 +1,37 @@
+--- ../src-base/minecraft/net/minecraft/item/ItemBed.java
++++ ../src-work/minecraft/net/minecraft/item/ItemBed.java
+@@ -5,19 +5,23 @@
+ import net.minecraft.block.SoundType;
+ import net.minecraft.block.state.IBlockState;
+ import net.minecraft.creativetab.CreativeTabs;
++import net.minecraft.entity.Entity;
```
Comment (shadowfacts) [inline patches/minecraft/net/minecraft/item/ItemBed.java.patch]: Once again, no imports

File: patches/minecraft/net/minecraft/item/ItemBed.java.patch [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+ import net.minecraft.util.SoundCategory;
++import net.minecraft.util.SoundEvent;
+ import net.minecraft.util.math.BlockPos;
+ import net.minecraft.util.math.MathHelper;
+ import net.minecraft.world.World;
+ 
+ public class ItemBed extends Item
+ {
+-    public ItemBed()
++    private Entity playerIn;
```
Comment (shadowfacts) [inline patches/minecraft/net/minecraft/item/ItemBed.java.patch]: 1. Why?
2. This isn't even used.

File: patches/minecraft/net/minecraft/item/ItemBlock.java.patch [hunk 7]
```diff
@@ -1,6 +1,14 @@
 --- ../src-base/minecraft/net/minecraft/item/ItemBlock.java
 +++ ../src-work/minecraft/net/minecraft/item/ItemBlock.java
-@@ -51,16 +51,8 @@
+@@ -16,6 +16,7 @@
+ import net.minecraft.util.EnumFacing;
+ import net.minecraft.util.EnumHand;
+ import net.minecraft.util.SoundCategory;
++import net.minecraft.util.SoundEvent;
```
Comment (shadowfacts) [inline patches/minecraft/net/minecraft/item/ItemBlock.java.patch]: Imports

File: patches/minecraft/net/minecraft/item/ItemBlockSpecial.java.patch [hunk 8]
```diff
@@ -0,0 +1,20 @@
+--- ../src-base/minecraft/net/minecraft/item/ItemBlockSpecial.java
++++ ../src-work/minecraft/net/minecraft/item/ItemBlockSpecial.java
+@@ -11,6 +11,7 @@
+ import net.minecraft.util.EnumFacing;
+ import net.minecraft.util.EnumHand;
+ import net.minecraft.util.SoundCategory;
++import net.minecraft.util.SoundEvent;
```
Comment (shadowfacts) [inline patches/minecraft/net/minecraft/item/ItemBlockSpecial.java.patch]: Imports again

File: patches/minecraft/net/minecraft/item/ItemDoor.java.patch [hunk 9]
```diff
@@ -0,0 +1,20 @@
+--- ../src-base/minecraft/net/minecraft/item/ItemDoor.java
++++ ../src-work/minecraft/net/minecraft/item/ItemDoor.java
+@@ -10,6 +10,7 @@
+ import net.minecraft.util.EnumFacing;
+ import net.minecraft.util.EnumHand;
+ import net.minecraft.util.SoundCategory;
++import net.minecraft.util.SoundEvent;
```
Comment (shadowfacts) [inline patches/minecraft/net/minecraft/item/ItemDoor.java.patch]: And again

**Comment by williewillus** [comment]
* Formatting is a mess
* Calling getBreakSound and co. will crash the dedicated server because those methods are SideOnly client so this PR is mostly broken
* What exactly is the purpose of this? You can already have your own sounds by changing the block's soundtype using setSoundType.

**Comment by Madsthunder** [comment]
I added this because of the fact that whenever you would call Block#getSoundType there are no variables to get or change a SoundType based on an IBlockState

**Comment by williewillus** [comment]
so make a state sensitive getSoundType

**Comment by williewillus** [comment]
In Block:

```Java
public SoundType getSoundType(IBlockState state) {
    return getSoundType();
}
```

patch vanilla to call that instead of the no args version. profit.

**Comment by Madsthunder** [comment]
Yes, that would be a good idea... But I was taking into account what ifs. For example: if somebody were to make a multipart block where each part could be broken individually then there would be no variables but 'Minecraft' to check what RayTraceResult the player/entity was looking at.

**Comment by Madsthunder** [comment]
Still, that IS a very good point

**Comment by williewillus** [comment]
uhhhh what? how is a multipart relevant to break sounds again?

**Comment by shadowfacts** [comment]
So pass in the World, BlockPos, and Entity as well as the IBlockState from which you can perform the ray trace yourself.

@williewillus each multipart could have it's own breaking sound

**Comment by williewillus** [comment]
i disagree with passing anything other than the state

**Comment by Madsthunder** [comment]
^^^ Yes

**Comment by williewillus** [comment]
mods can and will need to get block break sounds without a position or world

**Comment by williewillus** [comment]
either way break this down and rebuild it in a sensible manner please :P 
if you do have params that aren't state, mark them nullable

**Comment by Madsthunder** [comment]
Okay

**Pull request closed without merging**